samples/cpp/prompt_lookup_decoding_lm/CMakeLists.txt

```diff
@@ -16,6 +16,10 @@ set_target_properties(prompt_lookup_decoding_lm PROPERTIES
     INSTALL_RPATH_USE_LINK_PATH ON)
 target_compile_definitions(prompt_lookup_decoding_lm PRIVATE OPENVINO_TOKENIZERS_PATH="${OPENVINO_TOKENIZERS_PATH}")
 target_compile_features(prompt_lookup_decoding_lm PRIVATE cxx_std_17)
+if(APPLE)
+    # aligned deallocation is only available on macOS 10.14 or newer
+    target_compile_options(prompt_lookup_decoding_lm PRIVATE -fno-aligned-allocation)
```

Comment:
Why is it needed?

If you state that it's only available on macOS 10.14 or newer in the comment, why isn't that reflected in implementation? I'm not sure, but I think older macOS isn't supported, in that case the comment is misleading because it gives perception that older macOS is supported.

Reply:
We still support macOS>=12, see https://docs.openvino.ai/2024/about-openvino/release-notes-openvino/system-requirements.html

Reply:
You shouldn't keep a comment about 10.14 if it's not supported because it's too old

Reply:
I can remove, but the comment is still valid, because we set MACOSX_DEPLOYMENT_TARGET=10.12 on macOS x86

Reply:
Where do you set it?

Reply:
removed this fix at all, set target 11.0